### PR TITLE
chore(cli app scripts): update to 5.7.2 and adjust related files

### PR DESCRIPTION
**Pull request by Mohammer5**
- [x] Upgraded the `@dhis2/cli-app-scripts` package to 5.7.0
- [x] Changed `build/[es|cjs]/lib.js` to `./build/[es|cjs]/index.js`
- [x] Added the `exports` field to all workspaces' `package.json
- [x] Added locale files to sideeffects in the widgets and forms workspaces
- [x] Adjusted the storybook's babel config file
- [x] Figure out if the changes to get the icons library to work again are legit

When running `yarn build:icons` on master, the following files are being created:
```
packages/icons/build/es/lib.js
packages/icons/build/cjs/lib.js
```

When switching to this branch, this does not happen anymore until I added an `index.js` to the icons' src folder which imports `./react/index.js`

### CODE REVIEW

**Review by amcgee — APPROVED** [review]
Code change LGTM, obviously tests need to pass and I haven't tried running this locally

File: packages/icons/package.json [hunk 1]
```diff
@@ -2,8 +2,12 @@
...
     "version": "6.5.0",
     "description": "Icons used in the UI libs",
-    "main": "build/cjs/lib.js",
-    "module": "build/es/lib.js",
+    "main": "./build/cjs/index.js",
+    "module": "./build/es/index.js",
+    "exports": {
+        "import": "./build/es/index.js",
+        "require": "./build/cjs/index.js"
+    },
```
Comment (amcgee) [inline packages/icons/package.json:10]: @Mohammer5 I think you just want to update these (and the `entryPoints.lib` in `d2.config.js`) to `<base>/react/index.js` right?

Comment (Mohammer5) [reply packages/icons/package.json:10]: That produces the following output when building:

```
[build:icons] Build parameters: 
[build:icons]  - Mode : production
[build:icons] [WARNING] Invalid "main" field in package.json, expected "./build/cjs/index.js" (got "./build/cjs/react/index.js") 
[build:icons] [WARNING] Invalid "module" field in package.json, expected "./build/es/index.js" (got "./build/es/react/index.js") 
[build:icons] [WARNING] Invalid "pkg.exports.import" field in package.json, expected "./build/es/index.js" (got "./build/es/react/index.js") 
[build:icons] [WARNING] Invalid "pkg.exports.require" field in package.json, expected "./build/cjs/index.js" (got "./build/cjs/react/index.js") 
```

Comment (Mohammer5) [reply packages/icons/package.json:10]: Which causes the build command to become stale

Comment (varl) [reply packages/icons/package.json:10]: There is a bug here in the app-scripts: https://github.com/dhis2/app-platform/blob/master/cli/src/lib/validators/validatePackageExports.js#L42

The `entrypointBasename` at this point is `index.js`, and then when the expected exports `expectedESMExport` and `expectedCJSExport` are constructed, it is assumed that the `entrypointBasename` is always under `build/esm/index.js`.

In this case, the constructed path should be `build/esm/react/index.js` so the logic in app-scripts needs to be adjusted.

Comment (varl) [reply packages/icons/package.json:10]: Or, we say that there must be a `index.js` in the source directory, and don't support nested entry-points under `<source>/<other>/index.js`.

Comment (Mohammer5) [reply packages/icons/package.json:10]: I guess the `src/index.js` workaround in this PR should be ok for now, I'll create an issue (if you haven't done it already)

Comment (varl) [reply packages/icons/package.json:10]: https://github.com/dhis2/app-platform/pull/new/fix-validation-exports

**Comment by varl** [comment]
There is a funky test failure in the Transfer's LeftSide component when it imports the background color. I can resolve it manually, so it's definitely _there_, but for some reason loading the CJS fails in the test.

**Comment by varl** [comment]
Ah, I see:

`./core/src/AlertBar/AlertBar.stories.js: Error: styled-jsx/css: if you are getting this error it means that your `css` tagged template literals were not transpiled.`

**Comment by Mohammer5** [comment]
> There is a funky test failure in the Transfer's LeftSide component when it imports the background color. I can resolve it manually, so it's definitely there, but for some reason loading the CJS fails in the test.

I've seen that one as well. But I can't figure out why it's happening. It's never failed before, the imports seem correct. I was able to get the test to succeed again by renaming that file, run the test and then rename the file back to its original name. The second time the error occurred, I wasn't able to do anything about it (even after deleting all node module folders, clearing jest's and yarn's caches and then installing the packages again)

**Comment by varl** [comment]
The new `app-scripts build` command builds and copies everything from `src` to `build`, including all the test files and stories.

So when we run `app-scripts test` from the root, it finds all the test files in `src` and in `build` and runs all of them. The tests fail on the `build/**/*.test.js` files.

**Comment by varl** [comment]
The test is failing because we now copy all the source files to the build directory, and Jest is very eager when it comes to finding and running tests, so it also runs them from the `build/` directory. The tests don't necessarily work in their transpiled variants, so we get problems.

This fixes the `test` command in `app-scripts`: https://github.com/dhis2/app-platform/pull/521

**Comment by varl** [comment]
There is some strange interactions going on between Storybook's Webpack and Bable setup, where some sources are imported through the entry-points in the `build` folder, and some are imported through the `src` folder. The `build` imports are already transpiled so they work properly, but the `src` imports somehow end up not being transpiled by Storybook's Webpack+Bable combo, even after overriding them.

When I swap out e.g. `import { Button } from './Button.js'` to `import { Button } from '@dhis2/ui'` in `Button.stories.js`, Storybook starts working correctly.

**Comment by Mohammer5** [comment]
> When I swap out e.g. import { Button } from './Button.js' to import { Button } from '@dhis2/ui' in Button.stories.js, Storybook starts working correctly.

In my opinion importing from the built library is the actual correct implementation, because it doesn't require that storybook needs to be aware of any build steps required to get the component to work.
On the other side I can see why it's handy to just import the component via the relative path..
Given the current situation, I think I'd prefer changing the imports in all stories, if you agree (importing from `'@dhis2/...'`)

**Comment by varl** [comment]
@Mohammer5 That brings us to the next step.

Nothing in subpackages is allowed to import `@dhis2/ui`. We can either ESLint ignore in every Storybook file, or move the Storybook stuff outside of `core`.

**Comment by varl** [comment]
Alright, I figured out a way to override the Storybook Babel configuration with our necessary customization.

**Review by varl — APPROVED** [review]

**Pull request merged**